docs/docs/configuration/live.md

```diff
@@ -0,0 +1,20 @@
+---
+id: live
+title: Live View
+---
+
+Frigate has different live view options, some of which require [restream](restream.md) to be enabled.
+
+## Live View Options
+
+Live view options can be selected while viewing the live stream. The options are:
+
+| Source | Latency | Frame Rate                             | Resolution     | Audio                       | Requires Restream | Other Limitations  |
+| ------ | ------- | -------------------------------------- | -------------- | --------------------------- | ----------------- | ------------------ |
+| jsmpeg | low     | same as `detect -> fps`, capped at 10  | same as detect | no                          | no                | none               |
+| mp4    | high    | native                                 | native         | not yet                     | yes               | none               |
+| webrtc | low     | native                                 | native         | yes (depends on browser)    | yes               | requires host mode |
+
+### webRTC host mode
+
+webRTC works by creating a websocket connection on 2 random UDP ports (1 for video, 1 for audio). This means that frigate must be run with `network_mode: host` to support the webRTC live view mode.
```

Comment:
Well, you also have the option of exposing port 8555/tcp so that the udp ports aren't required. Therefore, not required to enable `network_mode: host` either.

But, ideally, go2rtc should let you configure the udp port range and home assistant supervisor as well:

https://community.home-assistant.io/t/add-support-for-port-ranges-in-add-on-config-yaml/375695

(please vote for it, if you can)

Reply:
I was going to make the same comment. There are a few things in the security settings we should think about as defaults here too.

Reply:
I'll give that a go tomorrow morning and see what others things make sense for a default config.

Reply:
@felipecrs after looking through this I don't think this is viable.

So the client makes a websocket connection to go2rtc (in this case I use nginx on location /live/webrtc/). Then the websocket returns with multiple candidates (being ip address +  udp ports and another being ip address + tcp port if specified in the config). The client then looks on that connection.

• bridge mode and port 8555 go2rtc sends the tcp candidate as 172.x.y.z 8555 which of course fails to resolve on the client
• bridge mode and port 8555 using the 
candidates:
    - stun:8555  # if you have dynamic public IP-address

sends the tcp candidate as your public ip address:8555 which works if 8555 is forwarded on your router.
• if I manually put a candidate as my own local IP address:
candidates:
    - 192.168.50.106:8555  # if you have dynamic public IP-address
 then it works

we could have `run as host or forward 8555 on your router` but I'm not sure if that is the right way to go
for now will reset this to previous

Reply:
Yeah, forwarding 8555 in your host is mandatory for **external** access. The same way it's also mandatory to have a publicly accessible IP for the auto-forwarded UDP ports anyway.

You can, however, most likely, get the Home Assistant IP address using its APIs and add it to the candidates, so it looks like:

```
candidates:
    - <ha-ip>:8555
    - stun:8555
```

Reply:
The problem is I don't think there's an API to add a candidate to the go2rtc config so we'd have to know that before starting which basically means the user has to bind mount and override the go2rtc config file with that info

Reply:
I was thinking about using `bashio` as part of a `cont-init.d` script, and perhaps using `tempio` to properly fill the `go2rtc.yaml` file.

But nevertheless it does not sound too bad to allow the user to set his own go2rtc config. After all the user may want to use go2rtc for other things as well.

Reply:
> Also that would not work well for the addon users either.

Unless you read this file from the /config directory, the same way you do for `frigate.yaml`.

Reply:
yes but then that would require the user to create one to even boot frigate up after upgrading to this new version

Reply:
Right, frigate could start go2rtc instead of s6 but that seems that it may have other complications, unless I am missing something and frigate could start, create a config file, and then s6 starts go2rtc

I was thinking maybe could just have a `LOCAL_HOST_NAME` or something environment variable which the user could set to support this.

Reply:
That's not what I had in mind, I meant something like:

`/etc/cont-init.d/go2rtc.sh`:

```bash

go2rtc_config=$(python get_go2rtc_config.py)

echo "${go2rtc_config}" > /go2rtc.yaml
```

Then, when you start go2rtc, you use the `/go2rtc.yaml`.

The `get_go2rtc_config.py` would call the Frigate functions to render a full configuration file and it would extract the `go2rtc` key from it alone by using Python's YAML parser.

Reply:
Ah I see what you're saying now, yeah that could definitely work, will give that a go

Reply:
After looking more into it I think nesting the go2rtc config inside frigate.yaml is messy because any change in the go2rtc schema would require an update in our config.

However, your previous idea would be good. I think I could simply check for a file at `/config/go2rtc.yml` and if it is not empty use it otherwise use the built in default config

Reply:
> any change in the go2rtc schema would require an update in our config.

It would be nice if go2rtc could expose this schema so that we could reuse it programmatically rather than having to maintain our own. If so, we could also automate it's update. Another option would be to keep the Frigate's go2rtc schema loose (i.e. only document the keys that Frigate actually have to set).

> However, your previous idea would be good. I think I could simply check for a file at /config/go2rtc.yml (which the addon can simply map go2rtc.yaml to) and if it is not empty use it otherwise use the built in default config

Yeah. I think this is good enough, but perhaps `/config/go2rtc.yml` would not be the best name (go2rtc addon also looks for it). Perhaps `frigate-go2rtc.yml`?

Reply:
> Yeah. I think this is good enough, but perhaps `/config/go2rtc.yml` would not be the best name (go2rtc addon also looks for it). Perhaps `frigate-go2rtc.yml`?

Yeah good idea, just updated and it is working locally and remotely without host mode which is great. Will update the docs now.